aspnetcore/fundamentals/portable-object-localization.md

```diff
@@ -0,0 +1,218 @@
+
+*fr.po*
+
+```text
+#: Services/EmailService.cs:29
+msgid "Enter a comma separated list of email addresses."
+msgstr "Entrez une liste d'emails séparés par une virgule."
+
+#: Views/Email.cshtml:112
+msgid "The email address is \"{0}\"."
+msgid_plural "The email addresses are \"{0}\"."
+msgstr[0] "L'adresse email est \"{0}\"."
+msgstr[1] "Les adresses email sont \"{0}\""
+```
+
+This example uses the following syntax:
+
+- `#:`: A comment indicating the context of the string to be translated. The same string might be translated differently depending on where it is being used.
+- `msgid`: The untranslated string.
+- `msgstr`: The translated string.
+
+In the case of pluralization support, more entries can be defined.
+
+- `msgid_plural`: The untranslated plural string.
+- `msgstr[0]`: The translated string for the case 0.
+- `msgstr[N]`: The translated string for the case N.
+
+The PO file specification can be found [here](https://www.gnu.org/savannah-checkouts/gnu/gettext/manual/html_node/PO-Files.html).
+
+## Configuring PO file support in ASP.NET Core
+
+This example is based on an ASP.NET Core MVC application generated from a Visual Studio 2017 project template.
+
+### Referencing the package
+
+Add a reference to the `OrchardCore.Localization.Core` NuGet package. It is available on [MyGet](https://www.myget.org/) at the following package source: https://www.myget.org/F/orchardcore-preview/api/v3/index.json
+
+The *.csproj* file now contains a line similar to the following (version number may vary):
+
+[!code-xml[Main](localization/sample/POLocalization/POLocalization.csproj?range=9)]
+
+### Registering the service
+
+Add the required services to the `ConfigureServices` method of *Startup.cs*:
+
+[!code-csharp[Main](localization/sample/POLocalization/Startup.cs?name=snippet_ConfigureServices&highlight=4-21)]
+
+Add the required middleware to the `Configure` method of *Startup.cs*:
+
+[!code-csharp[Main](localization/sample/POLocalization/Startup.cs?name=snippet_Configure&highlight=15)]
+
+Add the following code to your Razor view of choice. *About.cshtml* is used in this example.
+
+[!code-cshtml[Main](localization/sample/POLocalization/Views/Home/About.cshtml)]
+
+An `IViewLocalizer` instance is injected and used to translate the text `"Hello world!"`.
+
+### Creating a PO file
+
+Create a file named *fr.po*, in your application root folder, containing the following:
+
+[!code-text[Main](localization/sample/POLocalization/fr.po)]
+
+This file stores both the string to translate and the French-translated string. Translations revert to their parent culture, if necessary. In this example, the *fr.po* file is used if the requested culture is `fr-FR` or `fr-CA`.
+
+### Testing the application
+
+Run your application, and navigate to the URL `/Home/About`. The text **Hello world!** is displayed.
+
+Navigate to the URL `/Home/About?culture=fr-FR`. The text **Bonjour le monde!** is displayed.
```

Comment:
Is there an easy way to set the culture on the test machine so you don't need to pass a QS to use the culture? What about folks that are on a French machine? Do they need to do this?

Reply:
Sebastien is using the query string as it's one of the 3 default ways to change it (browser, cookie and querystring).